### PR TITLE
Add migration for substrate#14084

### PR DESCRIPTION
**Pull request by agryaznov**
<del>Companion</del> Add migration v12 for https://github.com/paritytech/substrate/pull/14084

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3016379

**Review by pgherveou — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3042130

File: client/network/src/lib.rs [hunk 1]
```diff
@@ -89,13 +89,14 @@ impl BlockAnnounceData {
...
 	/// This will not check the signature, for this you should use [`BlockAnnounceData::check_signature`].
 	fn validate(&self, encoded_header: Vec<u8>) -> Result<(), Validation> {
-		let candidate_hash =
-			if let CompactStatement::Seconded(h) = self.statement.unchecked_payload() {
-				h
-			} else {
-				tracing::debug!(target: LOG_TARGET, "`CompactStatement` isn't the candidate variant!",);
-				return Err(Validation::Failure { disconnect: true })
-			};
+		let candidate_hash = if let CompactStatement::Seconded(h) =
```
Comment (agryaznov) [inline client/network/src/lib.rs:92]: this is `cargo +nightly fmt` change, not related to this PR

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3108545

**Review by athei — APPROVED** [review]

File: client/network/src/lib.rs (hunk 1 above)
Comment (athei) [reply client/network/src/lib.rs:92]: CI is failing in this file. I suggest to roll back those changes.

Comment (athei) [reply client/network/src/lib.rs:92]: CI is failing in this file. I suggest to roll back those changes.

**Comment by agryaznov** [comment]
bot merge

**Pull request merged**